### DISCUSSION TITLE
Method Illuminate\Database\Eloquent\Collection::getMorphClass does not exist.

### DISCUSSION THREAD

**Discussion by utqinadhif** [livewire/livewire#7132] [1. Help] [answered]
### Livewire version

3.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

public $sendToView;

$model = Model->get();
$this->sendToView = $model->groupBy('field');

### Additional information

if i dd($model->groupBy('field')), this work, but if i send to public var error Method Illuminate\Database\Eloquent\Collection::getMorphClass does not exist. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@utqinadhif Livewire doesn't support custom queries, such as group by, selects, etc. when assigning models to properties. So you should use a computed property instead. See here for more details https://livewire.laravel.com/docs/properties#eloquent-constraints-arent-preserved-between-requests

Hope this helps!

**Reply by utqinadhif** [reply]
My project is from livewire 2, when livewire release v3, i migrate my app to v3, always running well, but today after composer upgrade, my app give me an error,

So, my case is new update from livewire?

**Reply by hapidjus** [reply]
We experienced this issue today as well when going from v2 to v3. I cant find anything about this in the upgrade guide.

**Reply by utqinadhif** [reply]
> We experienced this issue today as well when going from v2 to v3. I cant find anything about this in the upgrade guide.


I dont know until now, just wait for solution, sorry

**Comment by razvan-p** [comment]
@utqinadhif Same issue here. Updated from 3.0.4 to 3.3.3. Until this will be resolved, I'm using this code and it works:

```
public array $sendToView;

$model = Model->get();
$this->sendToView = $model->groupBy('field')->all();
```

**Reply by Marcovandelindt** [reply]
Ran into the same issue today and this worked for me. Thanks!

**Reply by sebGH06** [reply]
all() works for me too ! thanks

**Reply by moizmasood10** [reply]
Was stuck on this for 2 hours, thanks a lot, finally worked! Livewire should definitely offer support for custom queries asap.

**Comment by Luoti** [comment]
I will try to figure this out if I ever have the time. Just as a reminder. 3.0.5 works for me, but 3.0.6 breaks my queries.

_Edited:_

My own problem was that I created a two levels deep collection, that I assigned to a property. [Here is the line](https://github.com/livewire/livewire/compare/v3.0.5...v3.0.6#diff-d51171442b46f7a5604b48bfa4eec6e73c1afaa8589839bfe8aeb7a89c6368e7R32) that throws the error for me. Because I had two level deep collection, the method extracted the first level and then tried to call the `getMorphClass` against the model, which happened to be the second level collection.

I fixed this by modifying my collection to be one level deep only.

**Reply by donald-ov** [reply]
Just had the same issue. Fixed it by doing ->collect() on the top level collection which transformed it from Eloquent to Support collection.

**Reply by t-omw** [reply]
> Just had the same issue. Fixed it by doing ->collect() on the top level collection which transformed it from Eloquent to Support collection.

Thanks!

**Comment by JanSztefek** [comment]
The solutions is use toArray() it works for me:

$this->categories_with_food = RestaurantFood::with('category')
->select('id', 'name', 'quantity', 'allergens', 'price', 'category_id')
->whereDate('date', $formattedDate)
->get()
->groupBy(function ($food) {
return $food->category->name; // Access the category relationship
})->toArray();

**Reply by yandearta** [reply]
Using toArray() fixed the issue for me, thanks!

**Reply by sebGH06** [reply]
with toArray() you lost every information about eloquent relationship, all() method works well !
